### ISSUE TITLE
Radio scans introduce lag spikes

### ISSUE THREAD

**Issue by tdragan** [aircrack-ng/rtl8812au#778]
environment: 
  ubuntu,popOS 20.04, 20.10 
kernel: 
  5.4, 5.8 
repo branch:
 5.6.4.2 (current default branch)
device:
archer T2U Nano AC 600
issue: 
In what seems random intervals, there are lag spikes taking about 9 seconds which block network activity. Following closely ping to any site it can be seen that spikes occur consistently. When Network Manager is disabled it doesn't happen, because scan are not initiated. If I explicitly run "wpa_cli scan" I can reproduce the problem even if NM is turned off, so its not NM's fault. So I guess its wpa_supplicant/driver issue. My other devices (android phones and windows 10) have clean ping so I don't think its router related. NM power management set to 2 without success which is expected as Network Manager is not a culprit here.

**Comment by raplin** [comment]
...that is how scanning works! ;-) the radio retunes across the whole ISM band to look for APs; you will not get any data transfer while it's happening

**Comment by tdragan** [comment]
Should it block all traffic? It's like 9 seconds of disconnect if I play some multiplayer game I will be kicked due to timeout. Doesn't seem normal to me

**Comment by raplin** [comment]
Yes; literally the radio in the wifi dongle has to retune one at a time to all the other frequencies (separately on 2.4 and 5.8), then send a Probe Request, then listen for responses, and repeat. During this time it is simply not tuned to the frequencies of the AP it's connected to and can't do any comms.

**Comment by raplin** [comment]
solution is just to not have card scan when it's associated, this is a wpa_supplicant setting

**Comment by tdragan** [comment]
I will try to find wpa setting. I have set fixed BSSID but it doesn't help (as some source indicate). Very informative - thanks.

**Comment by mrbaguettovic** [comment]
Facing the same problem, @tdragan what setting did you use?

**Comment by tdragan** [comment]
I changed the backend and seen no lag spikes since then. This is what I used https://wiki.archlinux.org/title/NetworkManager#Using_iwd_as_the_Wi-Fi_backend

Although this explanation is for arch Linux it worked on fedora without issues.

**Comment by mrbaguettovic** [comment]
Thank you, hereby my journey to fixing my internet has ended :)
